### PR TITLE
DRIVERS-2789 Convert Max Staleness Spec to Markdown

### PR DESCRIPTION
**Pull request by blink1073**
Please complete the following before merging:

- [x] Update changelog.
- [ ] Make sure there are generated JSON files from the YAML test files.
- [ ] Test changes in at least one language driver.
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded
  clusters, and serverless).

### CODE REVIEW

File: source/max-staleness/max-staleness-tests.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+```
+mongodb://host/?maxStalenessSeconds=120
+mongodb://host/?readPreference=primary&maxStalenessSeconds=120
+```
+
+### Specifying "no max staleness" with "-1"
+
+A MongoClient with this connection string has no max staleness, the same as if "maxStalenessSeconds" were omitted from
+the connection string:
```
Comment (prestonvasquez) [inline source/max-staleness/max-staleness-tests.md:29]: It looks like we impose a 120 character line limit that is not explicitly defined in the `migrate_to_md.py` script, is this behavior intended?

File: source/max-staleness/max-staleness-tests.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+```
+mongodb://host/?readPreference=secondary&maxStalenessSeconds=-1
+```
+
+This connection string is valid and does not raise an error:
+
+> <mongodb://host/?readPreference=primary&maxStalenessSeconds=-1>
```
Comment (prestonvasquez) [inline source/max-staleness/max-staleness-tests.md:37]: Should this be in a code block?

````suggestion
```
mongodb://host/?readPreference=primary&maxStalenessSeconds=-1>
```
````

File: source/max-staleness/max-staleness.md [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+> 'P.lastUpdateTime',
+> 'P.lastWriteDate', 'S staleness'\],
+>
+> \['60', '10', 'P and S respond', '60', '0', '60', '10', '20 seconds'\], \['70', '20', 'S responds', '70', '5', '60',
+> '10', '25 seconds'\], \['80', '30', 'P responds', '70', '5', '80', '30', '25 seconds'\]
+>
+> \]
+>
+> print Table(data).create_table()
+
```
Comment (prestonvasquez) [inline source/max-staleness/max-staleness.md:245]: This was a comment in the original code, are we intentionally keep it as a quote?

https://github.com/mongodb/specifications/blob/39add29e0a3b6721688845874383878b90811a12/source/max-staleness/max-staleness.rst?plain=1#L271-L283

```suggestion
```

File: source/max-staleness/max-staleness.md [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+> 'P.lastUpdateTime', 'P.lastWriteDate', 'S
+> staleness'\],
+>
+> \['50', 'Idle write', '50', '', '50', '', ''\], \['60', 'Idle write begins', '60', '', '50', '', ''\], \['60', 'Client
+> checks P and S', '60', '60', '50', '60', '10.5'\], \['60', 'Idle write completes', '60', '', '60', '', ''\],
+>
+> \]
+>
+> print Table(data).create_table()
+
```
Comment (prestonvasquez) [inline source/max-staleness/max-staleness.md:309]: This was a comment in the original code, are we intentionally keep it as a quote?

https://github.com/mongodb/specifications/blob/39add29e0a3b6721688845874383878b90811a12/source/max-staleness/max-staleness.rst?plain=1#L344-L357

```suggestion
```

File: source/max-staleness/max-staleness.md [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+## Supplemental
+
+Python scripts in this document's source directory:
+
+- \`test_max_staleness_spo.py\`: Uses `scipy.optimize` to determine worst-case accuracy of the staleness estimate in an
+  idle replica set.
+- \`test_staleness_estimate.py\`: Tests whether a client would correctly select a secondary from an idle replica set,
+  given a random distribution of values for maxStalenessSeconds, heartbeatFrequencyMS, lastWriteDate, and
+  lastUpdateTime.
```
Comment (prestonvasquez) [inline source/max-staleness/max-staleness.md:330]: The \` does not render correctly in MD, this doesn't seem intentional.

https://github.com/mongodb/specifications/blob/39add29e0a3b6721688845874383878b90811a12/source/max-staleness/max-staleness.rst?plain=1#L374

```suggestion
- `test_max_staleness_spo.py`: Uses `scipy.optimize` to determine worst-case accuracy of the staleness estimate in an
  idle replica set.
- `test_staleness_estimate.py`: Tests whether a client would correctly select a secondary from an idle replica set,
  given a random distribution of values for maxStalenessSeconds, heartbeatFrequencyMS, lastWriteDate, and
  lastUpdateTime.
```

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

File: source/max-staleness/max-staleness-tests.md (hunk 1 above)
Comment (blink1073) [reply source/max-staleness/max-staleness-tests.md:29]: The config is [here](https://github.com/mongodb/specifications/blob/39add29e0a3b6721688845874383878b90811a12/.pre-commit-config.yaml#L41)

**Review by prestonvasquez — APPROVED** [review]

**Pull request merged**